database_models.go

```diff
@@ -160,9 +161,49 @@ const (
 // the file name and which build it belongs to.
 type Artifact struct {
 	ArtifactID uint   `gorm:"primaryKey" json:"artifactId"`
-	BuildID    uint   `gorm:"not null;index:param_idx_build_id" json:"buildId"`
+	BuildID    uint   `gorm:"not null;index:artifact_idx_build_id" json:"buildId"`
 	Build      *Build `gorm:"foreignKey:BuildID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
 	Name       string `gorm:"not null" json:"name"`
 	FileName   string `gorm:"nullable" json:"fileName"`
 	Data       []byte `gorm:"nullable" json:"-"`
 }
+
+// TestResultSummary contains data about a single test result file.
+type TestResultSummary struct {
+	TestResultSummaryID uint      `gorm:"primaryKey" json:"testResultSummaryId"`
+	FileName            string    `gorm:"nullable" json:"fileName"`
+	ArtifactID          uint      `gorm:"not null;index:testresultsummary_idx_artifact_id" json:"artifactId"`
+	Artifact            *Artifact `gorm:"foreignKey:ArtifactID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
+	BuildID             uint      `gorm:"not null;index:testresultsummary_idx_build_id" json:"buildId"`
+	Build               *Build    `gorm:"foreignKey:BuildID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
+	Total               uint      `gorm:"not null" json:"total"`
+	Failed              uint      `gorm:"not null" json:"failed"`
+	Passed              uint      `gorm:"not null" json:"passed"`
+	Skipped             uint      `gorm:"not null" json:"skipped"`
+}
+
+// TestResultStatus is an enum of different states a test result can be in.
+type TestResultStatus string
+
+const (
+	// TestResultStatusSuccess means the test succeeded.
+	TestResultStatusSuccess TestResultStatus = "Success"
+	// TestResultStatusFailed means the test failed.
+	TestResultStatusFailed TestResultStatus = "Failed"
+	// TestResultStatusSkipped means the test was skipped.
+	TestResultStatusSkipped TestResultStatus = "Skipped"
+)
+
+// TestResultDetail contains data about a single test in a test result file.
+type TestResultDetail struct {
+	TestResultDetailID uint             `gorm:"primaryKey" json:"testResultDetailId"`
+	ArtifactID         uint             `gorm:"not null;index:testresultdetail_idx_artifact_id" json:"artifactId"`
+	Artifact           *Artifact        `gorm:"foreignKey:ArtifactID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
+	BuildID            uint             `gorm:"not null;index:testresultdetail_idx_build_id" json:"buildId"`
+	Build              *Build           `gorm:"foreignKey:BuildID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
+	Name               string           `gorm:"not null" json:"name"`
+	Message            null.String      `gorm:"nullable" json:"message" swaggertype:"string"`
+	StartedOn          *time.Time       `gorm:"nullable;default:NULL;" json:"startedOn" format:"date-time"`
+	CompletedOn        *time.Time       `gorm:"nullable;default:NULL;" json:"completedOn" format:"date-time"`
```

Comment:
Suggest using `null.Time` instead

```suggestion
	StartedOn          null.Time        `gorm:"nullable;default:NULL;" json:"startedOn" format:"date-time"`
	CompletedOn        null.Time        `gorm:"nullable;default:NULL;" json:"completedOn" format:"date-time"`

```